Target.cc

```diff
@@ -429,8 +429,10 @@ const char *Target::NameIP(char *buf, size_t buflen) const {
 
 /* This next version returns a static buffer -- so no concurrency */
 const char *Target::NameIP() const {
-  if (!nameIPBuf) nameIPBuf = (char *) safe_malloc(MAXHOSTNAMELEN + INET6_ADDRSTRLEN);
-  return NameIP(nameIPBuf, MAXHOSTNAMELEN + INET6_ADDRSTRLEN);
+  /* Add 3 characters for the hostname and IP string, hence we allocate 
+      (FQDN_LEN + INET6_ADDRSTRLEN + 3) octets */
+  if (!nameIPBuf) nameIPBuf = (char *) safe_malloc(FQDN_LEN + INET6_ADDRSTRLEN + 3);
```

Comment:
Oh right, my mistake I knew that, I should have done it in the first place. Thanks for your review! I will commit the modifications ASAP.